include/eve/traits/overload/default_behaviors.hpp

```diff
@@ -228,6 +239,80 @@ namespace eve
     }
   };
 
+  //====================================================================================================================
+  //! @addtogroup extensions
+  //! @{
+  //!   @struct logical_elementwise_callable
+  //!   @brief CRTP base class giving an EVE's @callable the logical elementwise function semantic
+  //!
+  //!   **Defined in Header**
+  //!
+  //!   @code
+  //!   #include <eve/traits/overload.hpp>
+  //!   @endcode
+  //!
+  //!   Logical elementwise functions in EVE behave as strict elementwise callables but with the added ability to convert
+  //!   the output to a common logical type.
+  //!
+  //!   @tparam Func          Type of current @callable being implemented.
+  //!   @tparam OptionsValues Type of stored options.
+  //!   @tparam Options       List of supported option specifications.
+  //!
+  //!   @see strict_elementwise_callable
+  //! @}
+  //====================================================================================================================
+  template< template<typename> class Func
+          , typename OptionsValues
+          , typename... Options
+          >
+  struct logical_elementwise_callable : strict_elementwise_callable<Func, OptionsValues, Options...>
+  {
+    using base_t = strict_elementwise_callable<Func, OptionsValues, Options...>;
+    using ignore = typename base_t::ignore;
+    using func_t = typename base_t::func_t;
+
+    template<typename T, typename c_t>
+    EVE_FORCEINLINE static constexpr c_t cvt(T e, as<c_t>)
+    {
+      if      constexpr (std::same_as<T, c_t>)                     return e;
+      else if constexpr (std::same_as<T, bool> || scalar_value<T>) return c_t{e};
+      else                                                         return detail::call_convert(e, as_element<c_t>{});
+    }
+
+    template<callable_options O, typename T, typename... Ts>
+    constexpr EVE_FORCEINLINE auto behavior(auto arch, O const& opts, T x0, Ts const&... xs) const
+    {
+      if constexpr (match_option<condition_key, O, ignore_none_>)
+      {
+        using cl_t = common_logical_t<T, Ts...>;
+
+        if constexpr (std::same_as<cl_t, bool>)
+        {
+          // treat boolean as a special case: if the output is a boolean then all inputs must be booleans too.
```

Comment:
I don't think this phrase makes sense to me. 
It is just so happens that bool is the weakest logical. Why do you need this check.

Reply:
Sounds good to me to avoid more complex inner code.